### PR TITLE
fix: Fixes issue in `avm/res/container-instance/container-group` when container memory is set to less than 1 GB. 

### LINKED ISSUE #3655: [AVM Module Issue]: Unable to create a container instance with 0.5 GiB memory
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/container-instance/container-group

### (Optional) Module Version

_No response_

### Description

I am trying to create a Container Instance with 0.5 GB memory. In the module it says that I should use the `json()` function to specify a decimal value for memory. When I do this, I get the following error  
```
{"code": "InvalidTemplate", "target": "/subscriptions/<subscriptionid>/resourceGroups/<resourcegroup>/providers/Microsoft.Resources/deployments/containerInstanceVerifiedModule", "message": "Deployment template validation failed: 'The provided value for the template parameter 'containers[0].properties.resources.requests.memoryInGB' is not valid. Expected a value of type 'Integer, Null', but received a value of type 'Float'. Please see https://aka.ms/arm-create-parameter-file for usage details.'.", "additionalInfo": [{"type": "TemplateViolation", "info": {"lineNumber": 1, "linePosition": 4169, "path": "properties.template.definitions.containerType.items.properties.properties.properties.resources.properties.requests.properties.memoryInGB.type"}}]}
```

I call the template using the following values:

```
module dnsForwarderContainerInstance 'br/public:avm/res/container-instance/container-group:0.2.1' = {
  name: 'containerInstanceVerifiedModule'
  params: {
    subnetId: resourceId('Microsoft.Network/virtualNetworks/subnets', 'vnet-test', 'snet-test')
    containers: [
      {
        name: 'ci-test'
        properties: {
          environmentVariables: []
          image: '<path-to-my-image>'
          ports: [
            {
              port: 53
              protocol: 'Udp'
            }
          ]
          resources: {
            requests: {
              cpu: 1
              memoryInGB: json('0.5')
            }
          }
        }
      }
    ]
    ipAddressPorts: [
      {
        port: 53
        protocol: 'Udp'
      }
    ]
    ipAddressType: 'Private'
    name: 'ci-test'
    location: location
  }
}
```

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by anderseide**
## Description

* Changed `memoryInGB` prop in type `containerType` from `int` to `string`
* Updated tests to use `string` instead of `int`
* Added test that uses memoryInGB value less than 1
  * Updated documentation to no longer recommend using `json('0.5')` when using value less than 1

Closes #3655 

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.container-instance.container-group](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.container-instance.container-group.yml) |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [X] Breaking changes and I have bumped the MINOR version in `version.json`.
  - [X] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/container-instance/container-group/README.md
Comment (anderseide) [reply avm/res/container-instance/container-group/README.md:1]: Should be OK now. Latest build badge: [avm.res.container-instance.container-group](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.container-instance.container-group.yml)

Comment (AlexanderSehr) [reply avm/res/container-instance/container-group/README.md:1]: Looks good. Thanks @anderseide.

I'll leave the rest of the review to @JPEasier :)

File: avm/res/container-instance/container-group/main.bicep
Comment (JPEasier) [inline avm/res/container-instance/container-group/main.bicep:288]: The resource documentation says

> memoryInGB 
>
> The memory limit in GB of this container instance. To specify a decimal value, use the json() function. 
>
> int or json decimal

https://learn.microsoft.com/en-us/azure/templates/microsoft.containerinstance/containergroups?pivots=deployment-language-bicep#resourcelimits

I'm not sure whether this might lead to misunderstandings?

might be good to use type `any` and add the description again?

@eriqua, @AlexanderSehr what do you think?

**Review by JPEasier — CHANGES_REQUESTED** [review]

**Comment by jtracey93** [comment]
Hey @anderseide ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Comment by anderseide** [comment]
> Hey @anderseide ,
> 
> Firstly, thanks for your work on this PR!
> 
> We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.
> 
> ### Changes to CI That Have Been Made That You Need To Take Action On
> * [feat: Add WAF Security PS Rule Config #3745](https://github.com/Azure/bicep-registry-modules/pull/3745)
>   
>   * Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
>   * **Action Required By You:**
>     
>     * Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
>       
>       * If any issues with the rules and you think it should apply, please ping/reach out to the core team
> 
> Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams
> 
> Thanks
> 
> Jack (AVM Core Team)

After a bit back and forth build pipeline is now green: [avm.res.container-instance.container-group](https://github.com/anderseide/avm-bicep-registry-modules/actions/workflows/avm.res.container-instance.container-group.yml)

File: avm/res/container-instance/container-group/main.bicep
Comment (JPEasier) [reply avm/res/container-instance/container-group/main.bicep:288]: > The resource documentation says
> 
> > memoryInGB
> > The memory limit in GB of this container instance. To specify a decimal value, use the json() function.
> > int or json decimal
> 
> https://learn.microsoft.com/en-us/azure/templates/microsoft.containerinstance/containergroups?pivots=deployment-language-bicep#resourcelimits
> 
> I'm not sure whether this might lead to misunderstandings?
> 
> might be good to use type `any` and add the description again?
> 
> @eriqua, @AlexanderSehr what do you think?

@Azure/avm-core-team I need a second pair of eyes.

**Review by JPEasier — DISMISSED** [review]
Hi @anderseide,

after a discussion, let's implement your suggestion with ‘string’.

thank you for your contribution!

**Comment by JPEasier** [comment]
@anderseide please look on the conflicts, than i will merge your branch 🚀 
Thank you!

**Comment by anderseide** [comment]
> @anderseide please look on the conflicts, than i will merge your branch 🚀 Thank you!

Should be fixed now

**Comment by anderseide** [comment]
> @anderseide please look on the conflicts, than i will merge your branch 🚀 
> 
> Thank you!

Should be fixed, and build pipeline is succeeding.

**Review by AlexanderSehr — APPROVED** [review]
Thanks @anderseide, great work 🚀 
Will merge on @JPEasier's behalf given his last comment :)

**Pull request merged**
